plugins/module_utils/vm.py

```diff
@@ -1206,13 +1236,15 @@ def ensure_present_or_set(cls, module, rest_client, module_path):
                 module, rest_client, vm_before, changed, disk_key
             )
         if called_from_vm_disk:
+            vm_before.vm_power_up(module, rest_client)
             vm_after, disks_after = cls.get_vm_by_name(module, rest_client)
             return (
                 changed,
                 disks_after,
                 dict(before=disks_before, after=disks_after),
                 vm_before.reboot,
             )
+        vm_before.vm_power_up(module, rest_client)
         return changed, vm_before.reboot
```

Comment:
I think it would be more clear to put the `vm_power_up` call before line 1238 and remove the other one. The call is made whether the `if` condition is satisfied or not.

```suggestion
        vm_before.vm_power_up(module, rest_client)
        if called_from_vm_disk:
            vm_after, disks_after = cls.get_vm_by_name(module, rest_client)
            return (
                changed,
                disks_after,
                dict(before=disks_before, after=disks_after),
                vm_before.reboot,
            )
        return changed, vm_before.reboot
```